skimage/data/__init__.py

```diff
@@ -48,6 +53,159 @@
            'shepp_logan_phantom',
            'stereo_motorcycle']
 
+legacy_data_dir = osp.abspath(osp.dirname(__file__))
+skimage_distribution_dir = osp.join(legacy_data_dir, '..')
+
+try:
+    import pooch
+    from pooch.utils import file_hash
+    has_pooch = True
+except ImportError:
+    has_pooch = False
+    import hashlib
+
+    def file_hash(fname):
+        # Versioned from pooch.utils
+        # Calculate the hash in chunks to avoid overloading the memory
+        chunksize = 65536
+        hasher = hashlib.sha256()
+        with open(fname, "rb") as fin:
+            buff = fin.read(chunksize)
+            while buff:
+                hasher.update(buff)
+                buff = fin.read(chunksize)
+        return hasher.hexdigest()
+
+
+def _fetch(data_filename):
+    # potential_local_file is the location of the file that would be shipped
+    # with scikit-image.
+    #
+    # Two things can cause the file to already be in the distribution:
+    # 1. The user is running it from the github repo, in which case
+    #    we should avoid a downloading from the internet
+    # 2. We chose to ship this file with skimage by default
```

Comment:
Make this a docstring? =)

Reply:
Why? This is an implementation detail. A docgring might be justified though explaining how we check local cache too and the design choices behind the two errors that can be raised

Reply:
ok i added a docstring that talks about what the function *does* but not *how* it does it.